src/PHPStan/PregMatchTypeSpecifyingExtension.php

```diff
@@ -78,15 +78,11 @@ public function specifyTypes(MethodReflection $methodReflection, StaticCall $nod
             && count($matchedType->getConstantArrays()) === 1
         ) {
             $matchedType = $matchedType->getConstantArrays()[0];
-            $matchedType = new ConstantArrayType(
-                $matchedType->getKeyTypes(),
-                array_map(static function (Type $valueType): Type {
-                    return TypeCombinator::removeNull($valueType);
-                }, $matchedType->getValueTypes()),
-                $matchedType->getNextAutoIndexes(),
-                [],
-                $matchedType->isList()
-            );
+            foreach ($matchedType as $type) {
+                if ($type->containsNull()) {
+                    // TODO trigger an error here that $methodReflection->getName() should not be used, or the pattern needs to make sure that $type->?? get key name is not nullable
```

Comment:
@staabm Not sure how to trigger or if even possible from a type specifying extension?

Reply:
to trigger errors we need to build a new rule. this rule needs todo the same array-shape parsing an react on the resulting type then